server/src/middleware/errorHandler.js

```diff
@@ -5,11 +5,12 @@ const notFound = (req, res, next) => {
 };
 
 const errorHandler = (err, req, res, next) => {
-  const statusCode = res.statusCode ? res.statusCode : 500;
-  res.status(statusCode);
-  res.json({
-    error: err.message || "Internal server error!",
-    stack: process.env.NODE_ENV === "production" ? null : err.stack,
+  const statusCode = res.statusCode || 500;
+  res.status(statusCode).json({
+    error: err.message || "Internal Server Error",
+    stack: process.env.NODE_ENV === "development" ? err.stack : undefined,
```

Comment:
Why return _undefined_ instead of _null_?

Reply:
i just learned that returning `undefined` is better practice since you don't reveal the error stack in the property name. also this version of the code looks more cleaner to me. i think this one is good to go.